app/components/Form/ProjectEmissionIntensityReportForm.tsx

```diff
@@ -0,0 +1,238 @@
+
+  const handleChange = (
+    formChangeObject,
+    changeData,
+    changeStatus: "staged" | "pending"
+  ) => {
+    // don't trigger a change if the form data is an empty object
+    if (changeData && Object.keys(changeData).length === 0) return;
+
+    const updatedFormData = {
+      ...formChangeObject.newFormData,
+      ...changeData,
+    };
+    updateEmissionsIntensityReportFormChange({
+      variables: {
+        input: {
+          id: formChangeObject.id,
+          formChangePatch: {
+            newFormData: updatedFormData,
+            changeStatus: changeStatus,
+          },
+        },
+      },
+      optimisticResponse: {
+        updateFormChange: {
+          formChange: {
+            id: formChangeObject.id,
+            newFormData: updatedFormData,
+            changeStatus: changeStatus,
+          },
+        },
+      },
+      debounceKey: formChangeObject.id,
+    });
+  };
+
+  return (
+    <>
+      {projectRevision.emissionIntensityReportFormChange.edges.length === 0 && (
+        <Button
+          onClick={() =>
+            addProjectEmissionsIntensityReport({
+              variables: {
+                input: {
+                  revisionId: projectRevision.rowId,
+                },
+              },
+            })
+          }
+          style={{ marginRight: "1rem" }}
+        >
+          Add TEIMP Agreement
+        </Button>
+      )}
+      {projectRevision.emissionIntensityReportFormChange.edges.length > 0 && (
+        <>
+          <header>
+            <h3>Emission Intensity Report</h3>
+            <UndoChangesButton
+              formChangeIds={[
+                emissionIntensityReportFormChange.rowId,
+                reportingRequirementFormChange.rowId,
+              ]}
+              formRefs={formRefs}
+            />
+            <SavingIndicator
+              isSaved={
+                !isUpdatingEmissionsIntensityReportFormChange &&
+                !isAddingEmissionsIntensityReportFormChange
+              }
+            />
+          </header>
+          <FormBase
+            id="TEIMP_ReportingRequirementForm"
+            ref={(el) =>
+              (formRefs.current[reportingRequirementFormChange.id] = el)
+            }
+            validateOnMount={
+              reportingRequirementFormChange?.changeStatus === "staged"
+            }
+            idPrefix="TEIMP_ReportingRequirementForm"
+            schema={emissionIntensityReportingRequirementSchema as JSONSchema7}
+            formData={reportingRequirementFormChange?.newFormData}
+            formContext={{
+              form: reportingRequirementFormChange?.newFormData,
+            }}
+            uiSchema={emissionIntensityReportingRequirementUiSchema}
+            onChange={(change) =>
+              handleChange(
+                reportingRequirementFormChange,
+                change.formData,
+                "pending"
+              )
+            }
+          />
+          <FormBase
```

Comment:
Is the plan for further iterations to ultimately combine these so they look more like the wireframe?